felix/calc/encapsulation_resolver_test.go

```diff
@@ -0,0 +1,419 @@
+// Copyright (c) 2022 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package calc_test
```

Comment:
I _think_ it's OK to just put this in `package calc` and then you won't need to use special reflect logic to get unexplored fields.

Reply:
Might be worth a try

Reply:
I only had to export one helper function to be used in the calc_graph fv, but it worked. Thanks!